### PR TITLE
Payload building

### LINKED ISSUE #20: (Geth) Improve block building
Block building requires a 1 second delay, we should work to change this to reduce latency of execution. 

See https://github.com/astriaorg/go-ethereum/blob/astria/grpc/execution/server.go#L70

### PR DESCRIPTION
**Pull request by sambukowski**
Removed the sleep that was required before `GetPayloadV1` by removing the goroutine in `miner.buildPayload()`.
resolves #20

### CODE REVIEW

File: miner/payload_building.go [hunk 1]
```diff
@@ -163,36 +163,31 @@ func (w *worker) buildPayload(args *BuildPayloadArgs) (*Payload, error) {
...
-				timer.Reset(w.recommit)
-			case <-payload.stop:
-				log.Info("Stopping work on payload", "id", payload.id, "reason", "delivery")
-				return
-			case <-endTimer.C:
-				log.Info("Stopping work on payload", "id", payload.id, "reason", "timeout")
-				return
+	// Setup the timer for re-building the payload. The initial clock is kept
+	// for triggering process immediately.
+	timer := time.NewTimer(0)
```
Comment (joroshiba) [inline miner/payload_building.go]: Why do we still need the timer and loop?

File: miner/payload_building.go [hunk 2]
```diff
@@ -163,36 +163,31 @@ func (w *worker) buildPayload(args *BuildPayloadArgs) (*Payload, error) {
...
+			timer.Reset(w.recommit)
+		case <-payload.stop:
+			log.Info("Stopping work on payload", "id", payload.id, "reason", "delivery")
+			return payload, nil
+		case <-endTimer.C:
+			log.Info("Stopping work on payload", "id", payload.id, "reason", "timeout")
+			return payload, nil
 		}
-	}()
-	return payload, nil
```
Comment (joroshiba) [inline miner/payload_building.go:197]: This function is supposed to return a payload still

Comment (sambukowski) [reply miner/payload_building.go:197]: The payload gets returned out of the cases in the loop.

File: miner/payload_building.go (hunk 1 above)
Comment (sambukowski) [reply miner/payload_building.go]: turns out we don't! i was able to remove both the timer and the loop and everything is still working the same

**Review by joroshiba — APPROVED** [review]

**Pull request merged**